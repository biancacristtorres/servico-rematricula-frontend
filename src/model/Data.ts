export class Data {
  public data!: Date;
  public horaInicio!: Date;
  public horaFim!: Date;
}
