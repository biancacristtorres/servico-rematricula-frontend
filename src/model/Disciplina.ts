import { SituacaoDaDisciplina } from '@/common/enum/SituacaoDaDisciplina';
import { Turma } from './Turma';

export class Disciplina {
  public codigo: number;
  public nome: string;
  public situacao: SituacaoDaDisciplina;
  public modulo: string;
  public eDisciplinaEletiva: boolean;
  public possuiEquivalenciaDoisPorUm!: boolean;
  public turmas: Turma[] = [];
  public turmasEquivalentes: Turma[] = [];
  public turmasEspeciais: Turma[] = [];

  constructor(codigo: number, nome: string, situacao: SituacaoDaDisciplina,
              modulo: string, eDisciplinaEletiva: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.situacao = situacao;
    this.modulo = modulo;
    this.eDisciplinaEletiva = eDisciplinaEletiva;
  }
}
