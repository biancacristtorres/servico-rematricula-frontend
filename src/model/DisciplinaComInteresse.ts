export class DisciplinaComInteresse {
  public codigo?: number;
  public nome?: string;
  public interesseManifestado?: boolean;

  constructor(codigo: number, nome: string, interesseManifestado: boolean) {
    this.codigo = codigo;
    this.nome = nome;
    this.interesseManifestado = interesseManifestado;
  }
}
