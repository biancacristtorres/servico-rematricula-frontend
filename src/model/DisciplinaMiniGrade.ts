export class DisciplinaMiniGrade {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
}
