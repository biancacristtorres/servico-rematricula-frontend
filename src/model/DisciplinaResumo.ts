import { StatusDaDisciplina } from '@/common/enum/StatusDaDisciplina';

export class DisciplinaResumo {
  public codigoDisciplina?: number;
  public nomeDisciplina?: string;
  public descricaoTurma?: string;
  public status?: StatusDaDisciplina;
  public ePorData?: boolean;
  public campus?: string;
  public descricaoTipoCargaHoraria?: string;
  public datas?: string[];
  public nomeDisciplinaEquivalente?: string;
}
