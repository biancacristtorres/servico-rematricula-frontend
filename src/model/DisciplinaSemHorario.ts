export class DisciplinaSemHorario {
  public nomeDisciplina?: string;
  public nomeDisciplinaEquivalente?: string;
}
