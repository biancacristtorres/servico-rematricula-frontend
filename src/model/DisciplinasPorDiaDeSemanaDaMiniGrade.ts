import { DisciplinaMiniGrade } from './DisciplinaMiniGrade';

export class DisciplinasPorDiaDeSemanaDaMiniGrade {
  [index: number]: DisciplinaMiniGrade[];
  public 2: DisciplinaMiniGrade[];
  public 3: DisciplinaMiniGrade[];
  public 4: DisciplinaMiniGrade[];
  public 5: DisciplinaMiniGrade[];
  public 6: DisciplinaMiniGrade[];
  public 7: DisciplinaMiniGrade[];
}
