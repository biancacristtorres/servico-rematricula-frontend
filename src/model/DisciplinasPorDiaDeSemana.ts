import { DisciplinaResumo } from './DisciplinaResumo';

export class DisciplinasPorDiaDeSemana {
  [index: number]: DisciplinaResumo[];
  public 2: DisciplinaResumo[];
  public 3: DisciplinaResumo[];
  public 4: DisciplinaResumo[];
  public 5: DisciplinaResumo[];
  public 6: DisciplinaResumo[];
  public 7: DisciplinaResumo[];
}
