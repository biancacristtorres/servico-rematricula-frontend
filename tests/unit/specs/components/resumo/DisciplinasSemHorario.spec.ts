import _ from 'lodash';
import {
   DisciplinasSemHorarioFactory,
} from '../../../factory/components/resumo/DisciplinasSemHorarioFactory';
import { DisciplinaSemHorario } from '@/model/DisciplinaSemHorario';

describe('Componente de Disciplinas Sem Horário', () => {

  const factory = new DisciplinasSemHorarioFactory();

  beforeEach(() => {
    factory.prepararContainerDeInjecaoDaApi();
    factory.dadoQueAApiDeResumoRetorneDisciplinasSemHorarioComSucesso();
  });

  afterEach(() => {
    factory.restaurarContainerDeInjecaoDaApi();
  });

  it('snapshot', async (done) => {

    await factory.montarComponente();

    expect(factory.componente).toMatchSnapshot();

    done();
  });

  it('Quando uma disciplina não possui equivalente ' +
    'deve exibir somente o nome de uma disciplina', async (done) => {
    const disciplina = new DisciplinaSemHorario();
    disciplina.nomeDisciplina = 'Disciplina Da Grade';

    await factory.montarComponente();
    const atual = factory.vueInstance.formatarNomeDisciplina(disciplina);

    const esperado = 'Disciplina Da Grade';
    expect(atual).toEqual(esperado);

    done();
  });

  it('Quando uma disciplina possui uma equivalente com nome diferente da grade ' +
    'deve exibir o nome de ambas', async (done) => {
    const disciplina = new DisciplinaSemHorario();
    disciplina.nomeDisciplina = 'Disciplina Da Grade';
    disciplina.nomeDisciplinaEquivalente = 'Disciplina Equivalente';

    await factory.montarComponente();
    const atual = factory.vueInstance.formatarNomeDisciplina(disciplina);

    const esperado = 'Disciplina Equivalente (Eq. à Disciplina Da Grade)';
    expect(atual).toEqual(esperado);

    done();
  });
});
