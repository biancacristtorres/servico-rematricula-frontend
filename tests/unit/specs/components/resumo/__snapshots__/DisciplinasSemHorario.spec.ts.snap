// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de Disciplinas Sem Horário snapshot 1`] = `
<div class="container  box pa-0 pb-2">
  <!---->
  <div class="layout row wrap fill-width">
    <div class="fill-width title">
      <div class="caption pt-2 pb-2 mt-1 mb pl-4"><span class="subheading"> Disciplinas sem horário </span></div>
      <hr class="v-divider theme--light">
      <div class="v-list lista v-list--dense theme--light">
        <div>
          <div class="v-list__tile v-list__tile--avatar theme--light">
            <div class="v-list__tile__avatar">
              <div class="v-avatar" style="height: 40px; width: 40px;">
                <div class="v-responsive v-image icone">
                  <div class="v-responsive__content"></div>
                </div>
              </div>
            </div>
            <div class="v-list__tile__content"><span class="body-1">Disciplina Equivalente 1 (Eq. à Disciplina da Grade 1)</span></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
